src/units.rs

```diff
@@ -0,0 +1,109 @@
+use std::ops::{Div, Mul};
+
+use euclid::{Box2D, Point2D, Size2D, Vector2D};
+
+/// Coordinates in grid units (multiplies of the font size)
+pub struct Grid;
+/// Coordinates in pixel units
+pub struct Pixel;
+
+pub type GridVec<T> = Vector2D<T, Grid>;
+pub type PixelVec<T> = Vector2D<T, Pixel>;
+
+pub type GridSize<T> = Size2D<T, Grid>;
+pub type PixelSize<T> = Size2D<T, Pixel>;
+
+pub type GridPos<T> = Point2D<T, Grid>;
+pub type PixelPos<T> = Point2D<T, Pixel>;
+
+pub type GridRect<T> = Box2D<T, Grid>;
+pub type PixelRect<T> = Box2D<T, Pixel>;
+
+pub fn to_skia_point(pos: PixelPos<f32>) -> skia_safe::Point {
```

Comment:
Ah I see. These are type definitions. In that case maybe an extension trait? Importing free functions that do this feels wrong